tachyon_core/src/ffi.rs

```diff
@@ -140,38 +141,92 @@ pub unsafe extern "C" fn tachyon_inserter_value_type(inserter: *const Inserter)
 }
 
 /// SAFETY: The caller is responsible for calling `tachyon_inserter_flush` after finishing all insertions.
+/// On error (not code 0), this returns an error in the `out` parameter.
+/// The caller is responsible for freeing the returned pointer in `out`.
+/// Error data can be freed by using the function `tachyon_error_free`.
 #[no_mangle]
 pub unsafe extern "C" fn tachyon_inserter_insert_integer64(
     inserter: *mut Inserter,
     timestamp: Timestamp,
     value: i64,
-) {
-    (*inserter).insert_integer64(timestamp, value);
+    out: *mut *mut c_void,
+) -> u8 {
+    let result = (*inserter).insert_integer64(timestamp, value);
+
+    match result {
+        Ok(_) => 0u8,
+        Err(tachyon_err) => {
+            let return_value = get_error_code(&tachyon_err);
+            *out = Box::into_raw(Box::new(tachyon_err)) as *mut c_void;
+            return_value
+        }
+    }
 }
 
 /// SAFETY: The caller is responsible for calling `tachyon_inserter_flush` after finishing all insertions.
+/// On error (not code 0), this returns an error in the `out` parameter.
+/// The caller is responsible for freeing the returned pointer in `out`.
+/// Error data can be freed by using the function `tachyon_error_free`.
 #[no_mangle]
 pub unsafe extern "C" fn tachyon_inserter_insert_uinteger64(
     inserter: *mut Inserter,
     timestamp: Timestamp,
     value: u64,
-) {
-    (*inserter).insert_uinteger64(timestamp, value);
+    out: *mut *mut c_void,
+) -> u8 {
+    let result = (*inserter).insert_uinteger64(timestamp, value);
+
+    match result {
+        Ok(_) => 0u8,
+        Err(tachyon_err) => {
+            let return_value = get_error_code(&tachyon_err);
+            *out = Box::into_raw(Box::new(tachyon_err)) as *mut c_void;
+            return_value
+        }
+    }
 }
 
 /// SAFETY: The caller is responsible for calling `tachyon_inserter_flush` after finishing all insertions.
+/// On error (not code 0), this returns an error in the `out` parameter.
+/// The caller is responsible for freeing the returned pointer in `out`.
+/// Error data can be freed by using the function `tachyon_error_free`.
 #[no_mangle]
 pub unsafe extern "C" fn tachyon_inserter_insert_float64(
     inserter: *mut Inserter,
     timestamp: Timestamp,
     value: f64,
-) {
-    (*inserter).insert_float64(timestamp, value);
+    out: *mut *mut c_void,
+) -> u8 {
+    let result = (*inserter).insert_float64(timestamp, value);
+
+    match result {
+        Ok(_) => 0u8,
+        Err(tachyon_err) => {
+            let return_value = get_error_code(&tachyon_err);
+            *out = Box::into_raw(Box::new(tachyon_err)) as *mut c_void;
+            return_value
+        }
+    }
 }
 
+// SAFETY: On error (not code 0), this returns an error in the `out` parameter.
```

Comment:
Super tiny nit, this is a normal comment and not a doc comment

Reply:
Oops lol, fixed